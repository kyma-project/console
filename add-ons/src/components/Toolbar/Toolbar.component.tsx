import React from 'react';
import { ActionBar } from 'fundamental-react';
import { LuigiContext } from './../../services/LuigiContext.service';

import AddNewConfigurationModal from '../Modals/AddNewConfigurationModal/AddNewConfigurationModal.container';

import { CORE } from '../../constants';
import { StyledActionBar } from './styled';

const Toolbar: React.FunctionComponent = () => {
  const { namespaceId: currentNamespace } = React.useContext(LuigiContext);
  const title = currentNamespace
    ? CORE.ADD_ONS_CONFIGURATION_TITLE
    : CORE.CLUSTER_ADD_ONS_CONFIGURATION_TITLE;
  const description = currentNamespace
    ? CORE.ADD_ONS_CONFIGURATION_DESCRIPTION
    : CORE.CLUSTER_ADD_ONS_CONFIGURATION_DESCRIPTION;

  return (
    <StyledActionBar>
      <ActionBar.Header title={title} description={description} />
      <ActionBar.Actions>
        <AddNewConfigurationModal />
      </ActionBar.Actions>
    </StyledActionBar>
  );
};

export default Toolbar;
