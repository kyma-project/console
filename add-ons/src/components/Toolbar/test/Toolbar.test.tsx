import React from 'react';
import Toolbar from './../Toolbar.component';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Toolbar', () => {
  let useContextMock: any;
  let originalUseContext: any;

  beforeEach(() => {
    useContextMock = React.useContext = jest.fn();
    originalUseContext = React.useContext;
  });

  afterEach(() => {
    React.useContext = originalUseContext;
  });

  it('Renders cluster addons title', () => {
    useContextMock.mockReturnValue({}); // namespaceId is not defined in cluster view

    const component = shallow(<Toolbar />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Renders namespaced addons title', () => {
    useContextMock.mockReturnValue({ namespaceId: 'namespace' });

    const component = shallow(<Toolbar />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
