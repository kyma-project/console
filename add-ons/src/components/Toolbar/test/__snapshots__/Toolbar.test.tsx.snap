// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Renders cluster addons title 1`] = `
<Styled(ActionBar)>
  <ActionBar.Header
    description="Create and manage cluster-wide addons configurations, which allows you to extend the Service Catalog with additional cluster-wide services."
    headingLevel={3}
    title="Cluster Addons Configuration"
  />
  <ActionBar.Actions>
    <AddNewConfigurationModalContainer />
  </ActionBar.Actions>
</Styled(ActionBar)>
`;

exports[`Toolbar Renders namespaced addons title 1`] = `
<Styled(ActionBar)>
  <ActionBar.Header
    description="Create and manage Namespace-scoped addons configurations, which allows you to extend the Service Catalog with additional Namespace-scoped services."
    headingLevel={3}
    title="Addons Configuration"
  />
  <ActionBar.Actions>
    <AddNewConfigurationModalContainer />
  </ActionBar.Actions>
</Styled(ActionBar)>
`;
